api/ManifestApp/tests.py

```diff
@@ -1,3 +1,29 @@
 from django.test import TestCase
+from .models import *
 
-# Create your tests here.
+class HomepageTest(TestCase):
+    def test_url_exists_at_correct_location(self):
+        response = self.client.get("/ManifestApp/")
+        self.assertEqual(response.status_code, 200)
+
+class NodeTest(TestCase):
+    def setUp(self):
+        self.Node = NodeData.objects.create(VSN="A", name="A_name", gps_lat=50.00, gps_lan=50.00)
+
+        tag1 = Tag.objects.create(tag="t1")
+        self.Node.tags.set([tag1.pk])
+
+        compute1 = Hardware.objects.create(hardware="h1")
+        self.Node.computes.set([compute1.pk])
+
+    def test_node_creation(self):
+        N = self.Node
+        self.assertTrue(isinstance(N, NodeData))
+
+    def test_node_has_a_tag(self):
+        N = self.Node
+        self.assertEqual(N.tags.count(), 1)
+
+    def test_node_has_a_compute(self):
+        N = self.Node
+        self.assertEqual(N.computes.count(), 1)
```

Comment:
This is a good start.

Can we extend this with a test that exercises the public API? Something like:

```python
    def test_get_manifest(self):
        r = self.client.get("/ManifestApp/api/nodes/A/")
        self.assertEqual(r.status_code, 200)
        data = r.json()
        # check that data matches expected manifest format for this node and includes any repopulated compute, hardware, etc.
```